src/fft.jl

```diff
@@ -0,0 +1,576 @@
+using AbstractFFTs
+using LinearAlgebra
+using Dagger: DArray, @spawn, InOut, In
+using MPI
+
+struct FFT end
+struct RFFT end
+struct IRFFT end
+struct IFFT end
+struct FFT! end
+struct RFFT! end
+struct IRFFT! end
+struct IFFT! end
+
+export FFT, RFFT, IRFFT, IFFT, FFT!, RFFT!, IRFFT!, IFFT!, fft, ifft
+abstract type Decomposition end
+struct Pencil <: Decomposition end
+struct Slab <: Decomposition end
+
+function plan_transform(transform, A, dims; kwargs...)  
+        if transform isa RFFT
+            return plan_rfft(A, dims; kwargs...)
+        elseif transform isa FFT
+            return plan_fft(A, dims; kwargs...)
+        elseif transform isa IRFFT
+            return plan_irfft(A, dims; kwargs...)
+        elseif transform isa IFFT
+            return plan_ifft(A, dims; kwargs...)
+        elseif transform isa FFT!
+            return plan_fft!(A, dims; kwargs...)
+        elseif transform isa IFFT!
+            return plan_ifft!(A, dims; kwargs...)
+        else
+            throw(ArgumentError("Unknown transform type"))
```

Comment:
Please keep indentation to 4-spaces per indentation - you can use JuliaFormatter.jl to reformat your code.